scripts/perf/deploy_horicromtal/run_on_instance.sh

```diff
@@ -0,0 +1,72 @@
+#! /bin/bash
+
+set -x
+
+### /!\ This script assumes docker and docker compose are already installed on the host
+
+# Utility function to extract values from instance metadata
+extract_metadata() {
+  curl "http://metadata.google.internal/computeMetadata/v1/instance/attributes/$1" -H "Metadata-Flavor: Google"
```

Comment:
TOL can this output be stashed someplace to avoid the self-DOSing below?

Reply:
I don't understand this comment?

Reply:
I think he's saying that instead of curling for each of the many invocations of `extract_metadata()` it should be memoized somehow

Reply:
Not a big deal but yes, that 🙂 e.g. curl once to a file and then read the file repeatedly. I'm not sure it's doable since as the code is written now you actually are using different URLs, but if it's possible to use a higher-level URL that returns a key/value object structure it might be nice.